### PR TITLE
[breaking] Remove old gRPC Monitor service, in favor of new gRPC API based on pluggable monitor

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**
Remove the now obsolete gRPC `Monitor` service

- **Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**
Yes, even if we provide a better alternative, we are removing an API.

### CODE REVIEW

**Comment by fstasi** [comment]
blocked by [Zube #5626](https://zube.io/arduino/tooling-team/c/5626)

File: rpc/cc/arduino/cli/commands/v1/board.pb.go [hunk 1]
```diff
@@ -16,7 +16,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.26.0
-// 	protoc        v3.17.3
```
Comment (umbynos) [inline rpc/cc/arduino/cli/commands/v1/board.pb.go:19]: Is this change required?

File: docs/FAQ.md [hunk 2]
```diff
@@ -25,9 +25,6 @@ by this command is very limited and you may want to look for other tools if you
 There are many excellent serial terminals to chose from. On Linux or macOS, you may already have [screen][screen]
 installed. On Windows, a good choice for command line usage is Plink, included with [PuTTY][putty].
 
-Arduino CLI does provide a gRPC interface which offers the capability for powerful integration with custom monitors. See
-the [Monitor service documentation][monitor service].
```
Comment (umbynos) [inline docs/FAQ.md:29]: Is this still required? https://github.com/arduino/arduino-cli/blob/899dc91b3e2e12948badaffc25eca2cfaefa2eda/docs/FAQ.md?plain=1#L40

**Review by umbynos — CHANGES_REQUESTED** [review]
Added some comments

File: rpc/cc/arduino/cli/commands/v1/board.pb.go (hunk 1 above)
Comment (cmaglie) [reply rpc/cc/arduino/cli/commands/v1/board.pb.go:19]: no, I'll remove it

File: docs/FAQ.md (hunk 2 above)
Comment (cmaglie) [reply docs/FAQ.md:29]: We should replace this one earlier, I'm preparing a PR for this one only.

**Review by umbynos — APPROVED** [review]
